backend/dataall/core/resource_threshold/db/resource_threshold_repositories.py

```diff
@@ -0,0 +1,91 @@
+from dataall.core.resource_threshold.db.resource_threshold import ResourceThreshold
+from sqlalchemy import and_
+from datetime import date
+from dataall.base.config import config
+from dataall.base.db import exceptions
+from functools import wraps
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class ResourceThresholdRepository:
+    _RESOURCE_PATHS = {'nlq': 'modules.worksheets.features.max_count_per_day'}
+
+    @staticmethod
+    def _get_count_today(session, username, action_type):
+        amount = (
+            session.query(ResourceThreshold.count)
+            .filter(
+                and_(
+                    ResourceThreshold.username == username,
+                    ResourceThreshold.actionType == action_type,
+                    ResourceThreshold.date == date.today(),
+                )
+            )
+            .scalar()
+        )
+        return amount if amount else 0
+
+    @staticmethod
+    def _add_entry(session, username, action_type):
+        user_entry = ResourceThresholdRepository._get_user_entry(session, username, action_type)
+        if user_entry:
+            session.query(ResourceThreshold).filter(
+                and_(
+                    ResourceThreshold.username == username,
+                    ResourceThreshold.actionType == action_type,
+                )
+            ).update({ResourceThreshold.count: 1, ResourceThreshold.date: date.today()}, synchronize_session=False)
+            session.commit()
+        else:
+            action_entry = ResourceThreshold(username=username, actionType=action_type)
+            session.add(action_entry)
+            session.commit()
+
+    @staticmethod
+    def _increment_count(session, username, action_type):
+        session.query(ResourceThreshold).filter(
+            and_(
+                ResourceThreshold.username == username,
+                ResourceThreshold.actionType == action_type,
+                ResourceThreshold.date == date.today(),
+            )
+        ).update({ResourceThreshold.count: ResourceThreshold.count + 1}, synchronize_session=False)
+        session.commit()
+
+    @staticmethod
+    def _get_user_entry(session, username, action_type):
+        entry = (
+            session.query(ResourceThreshold)
+            .filter(and_(ResourceThreshold.username == username, ResourceThreshold.actionType == action_type))
+            .first()
+        )
+        return entry
+
+    @staticmethod
+    def check_invocation_count(action_type):
+        def decorator(func):
+            @wraps(func)
+            def wrapper(session, username, *args, **kwargs):
+                count = ResourceThresholdRepository._get_count_today(
+                    session=session, username=username, action_type=action_type
+                )
+                max_count = config.get_property(ResourceThresholdRepository._RESOURCE_PATHS[action_type], 10)
```

Comment:
I am not sure we should be declaring the thresholds in the ResourceThresholdRepository._RESOURCE_PATHS. 
PROS: from the threshold configuration we can know all the possible configurable thresholds; CONS: to use `check_invocation_count` from any module we need to update the core module.

We can pass the parameter value to check_invocation_count(action_type, max_count_parameter_path) or directly the max_count value and call `config.get_property(ResourceThresholdRepository._RESOURCE_PATHS[action_type], 10)` directly in the worksheets module

Reply:
agreed with the above - I will add as an input parameter the config path